src/lib/operators/validate.cpp

```diff
@@ -185,7 +192,27 @@ void Validate::_validate_chunks(const std::shared_ptr<const Table>& in_table, co
       } else {
         // Slow path - we are looking at multiple referenced chunks and need to get the MVCC data vector for every row.
         RowIDPosList temp_pos_list;
+        temp_pos_list.reserve(expected_number_of_valid_rows);
+
+        if (entirely_visible_chunks.empty()) {
+          // Check _is_entire_chunk_visible once for every chunk, even if we do not know if it is referenced or not.
+          // While this might introduce a small overhead in the case of many unreferenced chunks, it allows us to avoid
+          // a branch in the hot loop.
+          entirely_visible_chunks = std::vector<bool>(referenced_table->chunk_count(), false);
```

Comment:
As discussed offline, this assumes that all reference segments reference the same table.

Reply:
That was clear, wasn't it? The problem we had was that with each chunk iteration of the outer loop (L152) this single reference segment changes and the referenced table could also change. Since this branch is only taken during the first iteration when `entirely_visible_chunks` is empty, this change of the referenced table is not reflected.

Reply:
Ah, right. This assertion should help.